### PR TITLE
Add test for comments

### LINKED ISSUE #3204: Add a set of tests for comments
Should include:
- add
- remove
- ban
- unban

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Added tests related to the comment functionalities.

It should be testing the adding, removing, restoring of comments and also the banning and unbanning of users through the action popover that the comments have.

I'm new to writing tests, I tried to follow the cypress best practices in the docs. Every bit of feedback is very much welcomed.

Resolves #3204

### CODE REVIEW

**Review by chinins — COMMENTED** [review]
Looks good! A couple of points:
- I would suggest to use `data-test` so that if we have conflicts somewhere we can up the specificity with `data-cy` and have it kinda distinguished. I wrote a helper in `commands` file in cypress folder so that you could call it easier - `getBySel`. If you rebase on my `feature/3162-add-testing-framework-skeleton` branch, you will have it. I think that it's a good idea to have it the same everywhere.
- could you utilise more `timeout` instead of `wait`? It's more preferable. I know that in some cases it's not possible but we should try to use timeouts instead of wait as much as possible. I managed to get only 1 wait in so far...

**Review by arrenv — APPROVED** [review]
@ArmandoGraterol The tests look good to me. I had to increase the timeouts by double to get them to work on my end.

**Comment by ArmandoGraterol** [comment]
> @ArmandoGraterol The tests look good to me. I had to increase the timeouts by double to get them to work on my end.

@arrenv That's understandable! I've discovered while working on tests that the timeouts can be really hit or miss depending on your machine

File: cypress/e2e/3-comments.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+describe(
+  'Comments in action page',
+  { defaultCommandTimeout: 10000 },
```
Comment (chinins) [inline cypress/e2e/3-comments.js:3]: not necessary as it's already 10000 in the cypress config file

File: cypress/e2e/3-comments.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        'have.text',
+        commentText,
+      );
+    });
+
+    it('User with permissions can ban and unban users', () => {
+      cy.getBySel('comment-actions-button').last().click();
+      cy.getBySel('moderate-user-button').click();
+      cy.getBySel('moderate-user-confirm-button').click().wait(5000);
+      cy.getBySel('comment').last().should('have.css', 'color').and('eq', 'rgba(118, 116, 139, 0.8)');
```
Comment (chinins) [inline cypress/e2e/3-comments.js:25]: I like this testing expression 👍

**Review by chinins — APPROVED** [review]
Nice job - working well.

Could you please just rename the `3-comments` to `4-comments`. I have added create colony tests and it is `2-createColony` so actions moved to `3-actions`. It's just more logical and also needed for testing for create a colony to be in the beginning - thank you!

**Comment by arrenv** [comment]
> Could you please just rename the `3-comments` to `4-comments`. I have added create colony tests and it is `2-createColony` so actions moved to `3-actions`. It's just more logical and also needed for testing for create a colony to be in the beginning - thank you!

@chinins is numbering the best way to organise these, or is it arbitrary? It seems as though it could get messy. I noticed @ArmandoGraterol numbered extensions as `3-extensions` as well.

**Comment by chinins** [comment]
> > Could you please just rename the `3-comments` to `4-comments`. I have added create colony tests and it is `2-createColony` so actions moved to `3-actions`. It's just more logical and also needed for testing for create a colony to be in the beginning - thank you!
> 
> @chinins is numbering the best way to organise these, or is it arbitrary? It seems as though it could get messy. I noticed @ArmandoGraterol numbered extensions as `3-extensions` as well.

The tests will go into an alphabetic order based on file names - hence the numbering. It seems to be quite a standard practice. We should just not duplicate the numbers that the files start with I guess. I don't think that the order is very important, it's just that the claiming username & creating a colony should happen in the beginning.

**Comment by chinins** [comment]
@ArmandoGraterol, I just realised that your linting might not be working... Can you please check?
I see linting errors when I open the test files on my machine. E.g. you need to add linter exception every time you use wait:
`  // eslint-disable-next-line cypress/no-unnecessary-waiting`
Can you please check and fix them? We can get in touch if you need any help/have any questions about it. I went through lots of problems with it last week.

**Comment by ArmandoGraterol** [comment]
> @ArmandoGraterol, I just realised that your linting might not be working... Can you please check? I see linting errors when I open the test files on my machine. E.g. you need to add linter exception every time you use wait: ` // eslint-disable-next-line cypress/no-unnecessary-waiting` Can you please check and fix them? We can get in touch if you need any help/have any questions about it. I went through lots of problems with it last week.

Yeah, the linter is not running  (I wasn't even aware that there was a linter for the tests) so I'm gonna take a look at that

**Pull request merged**